### DISCUSSION TITLE
Is is possible to overide the .envfile?

### DISCUSSION THREAD

**Discussion by vesnikos** [nat-n/poethepoet#151] [Q&A]
having a task defined as eg bellow: 

```toms
[tool.poe.tasks.runserver]
envfile = ".env.dev"
```

is it possible to overide the envfile param manually on the cli? e.g.

`poe runserver --env-file .env.prod`

**Comment by nat-n** [comment]
Hi @vesnikos, 

That's an interesting idea, but I'm afraid there's currently no way to dynamically switch between envfiles for a task.

Hypothetically providing an envfile with an argument wouldn't fit very well with how poe works. The closest thing that would work would be if the poe CLI accepted an `--env-file` argument independently of the task being run, but I don't think this would generally be a very convenient feature to use.

However there are two ways that you could approach this:

1. What I would probably do is to define two tasks, each with their own configured envfile e.g. `serve` and `serve-prod`
2. You could also define one task that accepts an `--envfile` argument (which may have a default value), but then you would have to implement the envfile loading for yourself within the task somehow.
